src/generator.js

```diff
@@ -33,7 +33,7 @@ function generator(fn, argArray, options) {
             }
 
             beforeFormat.call(this);
-            return React.DOM.span(null, afterFormat.call(this, this.format()));
+            return React.DOM.span(componentProps, afterFormat.call(this, this.format()));
```

Comment:
could potentially filter/whitelist the componentProps–I don't think that would gain much though